cartoframes/data/services/geocoding.py

```diff
@@ -177,9 +176,53 @@ def _geocode_query(table, street, city, state, country, metadata):
         hash_column=HASH_COLUMN,
         hash_expression=hash_expression,
         geocode_expression=geocode_expression,
-        metadata_assignment=metadata_assignment
+        status_assignment=status_assignment
     )
 
+    return (query, status_columns)
+
+
+STATUS_FIELDS = {
+    'relevance': ('numeric', "(_g.metadata->>'relevance')::numeric"),
+    'precision': ('text', "_g.metadata->>'precision'"),
+    'match_types': ('text', "cdb_dataservices_client.cdb_jsonb_array_casttext(_g.metadata->>'match_types')"),
+    '*': ('jsonb', "_g.metadata")
+}
+
+
+def _status_column(column_name, field):
+    column_type, value = STATUS_FIELDS[field]
+    return (column_name, column_type, value)
+
+
+def _column_assignment(column_name, value):
+    return '{} = {},'.format(column_name, value)
+
+
+def _status_assignment(status):
+    status_assignment = ''
+    status_columns = []
+    if isinstance(status, dict):
+        # new style: define status assignments with dictionary
+        # {'column_name': 'status_field', ...}
+        # allows to assign individual status attributes to columns
+        invalid_fields = _list_difference(status.values(), STATUS_FIELDS.keys())
+        if any(invalid_fields):
+            raise ValueError("Invalid status fields {} valid keys are: {}".format(
+                invalid_fields,
+                STATUS_FIELDS.keys()))
```

Comment:
Using *.keys() in the message of an exception could cause some issues while testing.

If a test is added to cover this part since the output of keys() depends on the memory order and the Python version, the output is not deterministic, so the test will fail randomly.

I suggest using `STATUS_FIELDS_KEYS` with an array of the keys in a specific, and invariant, order

Reply:
:+1: done